src/backends/policy_engine/souffle/sql_policy.dl

```diff
@@ -0,0 +1,49 @@
+//-----------------------------------------------------------------------------
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//-----------------------------------------------------------------------------
+#ifndef SRC_BACKENDS_POLICY_ENGINE_SOUFFLE_SQL_POLICY_DL_
+#define SRC_BACKENDS_POLICY_ENGINE_SOUFFLE_SQL_POLICY_DL_
+
+#include "src/analysis/souffle/taint.dl"
+
+// OK: SELECT 42, 'foobar', null, true
+// OK: SELECT gender FROM people
+// FAIL: SELECT SSN FROM people
+//
+
+// Default owner associated with all tags related to SQL queries. 
+#define SQL_OWNER "sql"
+
+// A confidentiality tag used to capture that there restrictions.
+#define RESTRICTED_TAG "restricted"
+
+// Add restrictions on the access path.
+.decl restricted(result: AccessPath, tag: IntegrityTag)
```

Comment:
Another thing about `restricted`: unlike in the verifier, we will not have columns marked as implicitly having a tag. Rather, accesses to columns indicated like that in the policy will be wrapped in a `TagTransform` with no preconditions. So we do need to query for the `TagTransform` operations to find the initial tag state.

You can imagine it being something like this:

```
hasAppliedIntegrityTag(result, SQL_OWNER, tag) :- tagTransformResult(tranformNode, result), hasPolicyName(transformNode, "THIS_POLICY_NAME"), tagTransformAddsTag(transformNode, tag).
```

Where the above relations are populated from the attributes on the tag transform operation produced by the IR translator.

Reply:
Using `tagTransform` is how I started with, but wasn't sure why we would need it at all. IIUC, `restricted` is added to columns of _known_ tables. If we generate the same access path from `policy` and during the translation, I am not sure why we need the `tagTransform` operator at all.

Reply:
The reason why I'm using `TagTransform` for even statically-known columns is because I think it's against the spirit of the IR abstraction to have knowledge of how `AccessPath`s are generated from column names present in the frontend. But perhaps it would be a way to sidestep dealing with `TagTransform`s for an initial demo. It would, however, be adding logic for one special case whereas handling the `TagTransform` node would allow linking to policy rules for both columns and inner nodes of the query.